src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -841,6 +841,78 @@ namespace OpenLoco::Vehicles
         precedingDestComponent->setNextCar(source.id);
     }
 
+    // 0x004AF5E1
+    // esi: head
+    // returns nothing
+    void connectJacobsBogies(VehicleHead& head)
+    {
+        // if jacob's bogie is available at front the front body will have jacobsBogieAvailable
+        // if jacob's bogie is available at back the end body will have jacobsBogieAvailable
+        Vehicle train(head);
+        auto componentsFound = 0;
+        CarComponent previousCarComponent;
+        CarComponent secondPreviousCarComponent;
+        for (auto& car : train.cars)
+        {
+            // reset jacob's bogie at the front of the car
+            if (car.body->has38Flags(Flags38::jacobsBogieAvailable))
+            {
+                car.front->objectSpriteType = 0xFF;
+                car.back->objectSpriteType = 0xFF;
+                car.body->objectSpriteType = 0xFF;
+                auto frontBogieOfNext = car.body->nextVehicleComponent();
+                if (frontBogieOfNext == nullptr)
+                {
+                    throw Exception::RuntimeError("connectJacobsBogies frontBogieOfNext was unexpectedly nullptr");
+                }
+
+                CarComponent nextComponent = CarComponent(frontBogieOfNext);
+                auto o = ObjectManager::get<VehicleObject>(car.front->objectId);
```

Comment:
all of these objects are I'm sure the same but the assembly used nextComponent.front->objectId here.

Reply:
Deliberate change. I'll add a comment.

Reply:
If we are changing it we may as well use the same object for all of this and get rid of o2,o3, etc. just get the object before the first if